### PR TITLE
Change replica set detection for localhost to a single endpoint

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Change replica set detection for localhost to a single endpoint

## Motivation and Context
Replica set deployment improvements

## How to test this PR?
You need m3 deployment stage to test this PR

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#48eb4c8d-9949-48dd-a583-4dcb51c20615) |
| mint-fs.sh | :x: [more...](#3432a079-3ee0-42d8-85c0-f834d21a092c) |
<a name="3432a079-3ee0-42d8-85c0-f834d21a092c"></a>
### 8692-31f5c7f/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:32303
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8ddad81a8b34:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 14 seconds
{
  "name": "awscli",
  "duration": 312201,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "HTTPConnectionPool(host='72.28.97.55', port=32303): Max retries exceeded with url: /awscli-mint-test-bucket-18085 (Caused by ConnectTimeoutError(<botocore.awsrequest.AWSHTTPConnection object at 0x7f393e894748>, 'Connection to 72.28.97.55 timed out. (connect timeout=60)'))"
}

Executed 0 out of 13 tests successfully.
```
<a name="48eb4c8d-9949-48dd-a583-4dcb51c20615"></a>
### 8692-31f5c7f/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:32477
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 6b0eed1a9632:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 9 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 6 seconds
(4/13) Running mc tests ... done in 1 minutes and 6 seconds
(5/13) Running minio-go tests ... FAILED in 3 minutes and 14 seconds
{
  "args": {},
  "duration": 80342,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#911617e9-b451-4211-96b4-bf03ab01fa23) |
| mint-xl.sh | :x: [more...](#20f6e36f-47e0-411e-bcfc-9386ba695ceb) |
| mint-large-bucket.sh | :x: [more...](#9429e4af-1787-40a2-bc5c-53d1d88fdafb) |
<a name="911617e9-b451-4211-96b4-bf03ab01fa23"></a>
### 8692-2590667/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:32226
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5ba7567ba76b:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 14 seconds
{
  "name": "awscli",
  "duration": 12609,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}

Executed 0 out of 13 tests successfully.
```
<a name="9429e4af-1787-40a2-bc5c-53d1d88fdafb"></a>
### 8692-2590667/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:32222
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 13281dfe73c5:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 19 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 5 seconds
(4/13) Running mc tests ... done in 51 seconds
(5/13) Running minio-go tests ... FAILED in 2 minutes and 43 seconds
{
  "args": {},
  "duration": 54412,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
<a name="20f6e36f-47e0-411e-bcfc-9386ba695ceb"></a>
### 8692-2590667/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:30674
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp fd6edce64fb1:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 13 seconds
{
  "name": "awscli",
  "duration": 11203,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}

Executed 0 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#16917416-ec1a-4ab1-a6a5-c3f57ec56a7f) |
| mint-large-bucket.sh | :x: [more...](#461bf184-ad16-45f5-95e8-9b39b1027019) |
| mint-fs.sh | :x: [more...](#730170be-d5c6-4232-965b-43987d3108a7) |
<a name="16917416-ec1a-4ab1-a6a5-c3f57ec56a7f"></a>
### 8692-3be135a/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:31116
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4198953d37f0:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 8 seconds
{
  "name": "awscli",
  "duration": 5468,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}

Executed 0 out of 13 tests successfully.
```
<a name="461bf184-ad16-45f5-95e8-9b39b1027019"></a>
### 8692-3be135a/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:30143
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a34c195cee05:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 8 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 6 seconds
(4/13) Running mc tests ... done in 1 minutes and 19 seconds
(5/13) Running minio-go tests ... FAILED in 3 minutes and 51 seconds
{
  "args": {},
  "duration": 113976,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
<a name="730170be-d5c6-4232-965b-43987d3108a7"></a>
### 8692-3be135a/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:30897
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 293f67035113:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 11 seconds
{
  "name": "awscli",
  "duration": 309136,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "HTTPConnectionPool(host='72.28.97.55', port=30897): Max retries exceeded with url: /awscli-mint-test-bucket-8934 (Caused by ConnectTimeoutError(<botocore.awsrequest.AWSHTTPConnection object at 0x7eff30f720b8>, 'Connection to 72.28.97.55 timed out. (connect timeout=60)'))"
}

Executed 0 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by kannappanr — APPROVED** [review]
LGTM

File: cmd/storage-rest-common.go [hunk 1]
```diff
@@ -17,7 +17,7 @@
 package cmd
 
 const (
-	storageRESTVersion       = "v11"
+	storageRESTVersion       = "v12"
```
Comment (vadmeste) [inline cmd/storage-rest-common.go:20]: Why we are increasting storage version @harshavardhana ?

**Review by vadmeste — COMMENTED** [review]
One comment

File: cmd/storage-rest-common.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/storage-rest-common.go:20]: Because we forgot to change it in last PR #8690

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#816a1527-0938-4e93-be3b-19816e8dcbd2) |
<a name="816a1527-0938-4e93-be3b-19816e8dcbd2"></a>
### 8692-c4ce800/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:30167
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7ce85797c34e:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 14 seconds
{
  "name": "awscli",
  "duration": 312221,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "HTTPConnectionPool(host='72.28.97.55', port=30167): Max retries exceeded with url: /awscli-mint-test-bucket-29794 (Caused by ConnectTimeoutError(<botocore.awsrequest.AWSHTTPConnection object at 0x7fc3f742a128>, 'Connection to 72.28.97.55 timed out. (connect timeout=60)'))"
}

Executed 0 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#b11836b6-4b15-41b0-9a87-37b7fb5cfa1c) |
| mint-large-bucket.sh | :x: [more...](#2e662ea6-c68f-4054-b512-fb13cf9a8ad0) |
| mint-compression-fs.sh | :x: [more...](#f366f845-a11d-470a-946a-fdb5e98633ca) |
| mint-dist-xl.sh | :x: [more...](#caceceb2-1f7b-4354-8882-3816c3788846) |
<a name="f366f845-a11d-470a-946a-fdb5e98633ca"></a>
### 8692-e3c198f/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:32704
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c0f28ea07de6:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 12 seconds
{
  "name": "awscli",
  "duration": 310942,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "HTTPConnectionPool(host='72.28.97.55', port=32704): Max retries exceeded with url: /awscli-mint-test-bucket-592 (Caused by ConnectTimeoutError(<botocore.awsrequest.AWSHTTPConnection object at 0x7f59a05b5048>, 'Connection to 72.28.97.55 timed out. (connect timeout=60)'))"
}

Executed 0 out of 13 tests successfully.
```
<a name="2e662ea6-c68f-4054-b512-fb13cf9a8ad0"></a>
### 8692-e3c198f/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30531
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f71b914640fc:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 5 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 6 seconds
(4/13) Running mc tests ... done in 56 seconds
(5/13) Running minio-go tests ... FAILED in 3 minutes and 20 seconds
{
  "args": {},
  "duration": 94990,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
<a name="b11836b6-4b15-41b0-9a87-37b7fb5cfa1c"></a>
### 8692-e3c198f/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:32671
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp dcfe194b70f9:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 8 seconds
{
  "name": "awscli",
  "duration": 306412,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "HTTPConnectionPool(host='72.28.97.55', port=32671): Max retries exceeded with url: /awscli-mint-test-bucket-6720 (Caused by ConnectTimeoutError(<botocore.awsrequest.AWSHTTPConnection object at 0x7f5a6c743748>, 'Connection to 72.28.97.55 timed out. (connect timeout=60)'))"
}

Executed 0 out of 13 tests successfully.
```
<a name="caceceb2-1f7b-4354-8882-3816c3788846"></a>
### 8692-e3c198f/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:30105
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5dbf42ca4828:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 10 seconds
{
  "name": "awscli",
  "duration": 308395,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "HTTPConnectionPool(host='72.28.97.55', port=30105): Max retries exceeded with url: /awscli-mint-test-bucket-133 (Caused by ConnectTimeoutError(<botocore.awsrequest.AWSHTTPConnection object at 0x7f2cf9d95710>, 'Connection to 72.28.97.55 timed out. (connect timeout=60)'))"
}

Executed 0 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Pull request merged**